src/jit/codegenxarch.cpp

```diff
@@ -8851,41 +9048,37 @@ void CodeGen::genPutArgStk(GenTreePtr treeNode)
 //
 // Arguments
 //    treeNode      - the GT_PUTARG_STK node
-//    baseVarNum    - the variable number relative to which to put the argument on the stack.
-//                    For tail calls this is the baseVarNum = 0.
-//                    For non tail calls this is the outgoingArgSpace.
 //
-// Return value:
-//    None
+// Notes:
+//    In the case of fixed out args, the caller must have set m_stkArgVarNum to the variable number
+//    corresponding to the argument area (where we will put the argument on the stack).
+//    For tail calls this is the baseVarNum = 0.
+//    For non tail calls this is the outgoingArgSpace.
+
 //
-void CodeGen::genPutStructArgStk(GenTreePtr treeNode, unsigned baseVarNum)
+void CodeGen::genPutStructArgStk(GenTreePutArgStk* putArgStk)
 {
-    assert(treeNode->OperGet() == GT_PUTARG_STK);
-    assert(baseVarNum != BAD_VAR_NUM);
-
-    var_types targetType = treeNode->TypeGet();
+    var_types targetType = putArgStk->TypeGet();
 
     if (varTypeIsSIMD(targetType))
     {
-        regNumber srcReg = genConsumeReg(treeNode->gtGetOp1());
+        regNumber srcReg = genConsumeReg(putArgStk->gtGetOp1());
         assert((srcReg != REG_NA) && (genIsValidFloatReg(srcReg)));
-        getEmitter()->emitIns_S_R(ins_Store(targetType), emitTypeSize(targetType), srcReg, baseVarNum,
-                                  treeNode->AsPutArgStk()->getArgOffset());
+        genStoreRegToStackArg(targetType, srcReg, 0);
```

Comment:
Because codegen will now set the initial arg offset in m_stkArgOffset when it encounters the node.

---
In reply to: [79509737](https://github.com/dotnet/coreclr/pull/7252#discussion_r79509737) [](ancestors = 79509737)